src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,28 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        lock.readLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+            lock.readLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
```

Comment:
You missed locking the read lock and unlocking it with a try-finally block for the `get` method. Add the following code: `lock.readLock().lock();` before returning the element and `finally { lock.readLock().unlock(); }` block after that. 🚀